File: educational-system-consumer-student-9001/src/main/java/edu/hubu/consumerstudent/service/IStudentScoreService.java
package edu.hubu.consumerstudent.service;

import edu.hubu.commons.model.CommonResult;
import edu.hubu.commons.model.student.StudentScore;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.*;

/**
 * @author moonlan
 * date 2021/2/1 下午4:54
 */
@SuppressWarnings(value = "unused")
@FeignClient(name = "student-provider-service")
public interface IStudentScoreService {

    @GetMapping("/studentscore/{id}")
    CommonResult selectOne(@PathVariable("id") String id);

    @GetMapping("/studentscore/all")
    CommonResult selectAll(@RequestParam(value = "orderBy", defaultValue = "studentscoreId") String orderBy);

    @GetMapping("/studentscore/")
    CommonResult selectAllPage(
            @RequestParam(value = "currentPage", defaultValue = "1") Integer currentPage,
            @RequestParam(value = "pageSize", defaultValue = "5") Integer pageSize,
            @RequestParam(value = "orderBy", defaultValue = "studentscoreId") String orderBy
    );

    @PostMapping("/studentscore/select-like")
    CommonResult selectLikePage(
            @RequestParam(value = "currentPage", defaultValue = "1") Integer currentPage,
            @RequestParam(value = "pageSize", defaultValue = "5") Integer pageSize,
            @RequestParam(value = "orderBy", defaultValue = "studentscoreId") String orderBy,
            @RequestBody edu.hubu.commons.mybatis.model.StudentScore studentscore
    );

    @PostMapping("/studentscore/")
    CommonResult save(@RequestBody StudentScore studentscore);

    @PutMapping("/studentscore/{id}")
    CommonResult update(@PathVariable("id") String id, @RequestBody StudentScore studentscore);


    @DeleteMapping("/studentscore/delete/{id}")
    CommonResult deleteById(@PathVariable("id") String id);
}

File: educational-system-commons/src/main/java/edu/hubu/commons/service/student/IStudentScoreService.kt
package edu.hubu.commons.service.student

import edu.hubu.commons.model.student.StudentScore;

/**
* @author moonlan
* date 2021/1/25 下午2:07
*/
interface IStudentScoreService {
    fun save(t: StudentScore): StudentScore
    fun deleteById(id: String): Boolean
    fun updateById(id: String, t: StudentScore): StudentScore
    fun selectOneById(id: String): StudentScore
    fun selectAllOrderBy(order: String = "studentscoreId"): List<StudentScore>
    fun selectAllPageOrderBy(currentPage: Int = 1, pageSize: Int = 5, order: String = "studentscoreId"): List<StudentScore>
    fun selectLikePageOrderBy(currentPage: Int = 1, pageSize: Int = 5, order: String = "studentscoreId", t: edu.hubu.commons.mybatis.model.StudentScore): List<edu.hubu.commons.mybatis.model.StudentScore>
}
